lib/binary-buffer.js

```diff
@@ -0,0 +1,39 @@
+"use strict";
+
+module.exports = () => {
+    /**
+     * Converts a string into a buffer. Does not change the input if it is
+     * already a buffer.
+     *
+     * @param {(Buffer|string)} stringOrBuffer
+     * @return {(Buffer|string)}
+     */
+    function toBuffer(stringOrBuffer) {
+        if (!Buffer.isBuffer(stringOrBuffer)) {
+            stringOrBuffer = new Buffer(stringOrBuffer.toString(), "binary");
```

Comment:
I did find one test in which the `.toString()` method was having some influence. After removing `.toString()` in binary-buffer, this test in spec/lib/formatter/binary-formatter  breaks:

    it("essentially breaks with objects", () => {
         return formatterAsync({}).then((result) => {
             expect(Buffer.isBuffer(result)).toBe(true);                                                                                                                                                                                                                                                               
             expect(result.toString("binary")).toBe("[object Object]");
         });
    });

I was able to fix it by doing this: 

    it("essentially breaks with objects", () => {
        return formatterAsync({}).then(jasmine.fail, (err) => {                                                                                                                                                                                                                                                       
            expect(err.toString()).toBe("TypeError: First argument must be a string, Buffer, ArrayBuffer, Array, or array-like object.");
        }); 
    });

I just wanted to make sure that this would be an ideal solution, or if there would be a better alternative.